edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Http/middleware/WebSocketHandlingMiddleware.cs

```diff
@@ -78,17 +79,28 @@ async Task ProcessRequestAsync(HttpContext context, IWebSocketListener listener,
             var remoteEndPoint = new IPEndPoint(context.Connection.RemoteIpAddress, context.Connection.RemotePort);
 
             X509Certificate2 cert = await context.Connection.GetClientCertificateAsync();
+            IAuthenticator proxyAuthenticator = null;
 
             if (cert == null)
             {
-                var certExtractor = await this.httpProxiedCertificateExtractorProvider;
-                cert = (await certExtractor.GetClientCertificate(context)).OrDefault();
+                try
+                {
+                    var certExtractor = await this.httpProxiedCertificateExtractorProvider;
+                    cert = (await certExtractor.GetClientCertificate(context)).OrDefault();
```

Comment:
So, certExtractor.GetClientCertificate will throw if there is a cert in the header, but the sender module is not trusted somehow (bad token, no token, or moduleId invalid (not registered or different)), but if there is no cert in the header, then this will just return null. Right?
If so, can you please add a comment saying so in the code?